content/tutorials/aws-persistence.textile

```diff
@@ -0,0 +1,241 @@
+
+From the "AWS dashboard":http://console.aws.amazon.com/:
+
+# Click @Services@ from the top left menu
+# Select @Lambda@ (you can search in the toolbar if necessary)
+# Click @Create function@
+# Click @Author from Scratch@
+# Set your function name as @ably-persistence@
+# Click @Change default execution role@
+# Select @Python 3.9@ from the @Runtime@ drop down
+# Click @Use an existing role@ and choose @ably-persistence@ from the drop down
+# Click @Create function@
+
+You'll be presented with the function page with an empty "lambda_function.py" in your page:
+
+<a href="/images/tutorials/persistence/new-lambda.png" target="_blank">
+  <img src="/images/tutorials/persistence/new-lambda.png" style="width: 100%" alt="New lambda page">
+</a>
+
+Update @lambda_function.py@ with this code, remembering to add replace the bucket name with the one you created:
+
+```[python]
+import json, boto3, os, sys, uuid
+from urllib.parse import unquote_plus
+
+s3_client = boto3.client('s3')
+
+def lambda_handler(event, context):
+    
+    event_name = (event['name'])                   # Choosing name field
+    message = (event['data'])                      # Choosing data field
+    data_to_save = event_name + ' : ' + message    # Combining fields
+    print(data_to_save)                            # Checking combination
+    
+    bucket_name = "your-bucket-name-here"          # Target S3 Bucket
+    file_name = "test.txt"                         # Target file
+    lambda_path = "/tmp/" + file_name              # Path of file being download
+    s3_path = "output/" + file_name                # Path of file being uploaded
+    
+    file_object = open(lambda_path, 'a')           # Open downloaded file to append
+    file_object.write("\n" + data_to_save)         # Update downloaded file
+    file_object.close()                            # Save and close edited file
+    
+    s3 = boto3.resource("s3")
+    s3.meta.client.upload_file(lambda_path, bucket_name, file_name)
+```
+
+Once the webhook is set up this code will be invoked every time a message is published. When invoked it will take the data from the JSON Ably sends to Lambda, then open our @test.txt@, append the latest message, save the message and finally upload it back to S3.
+
+Now we're finished in AWS we can set up the Ably half of the workflow.
+
+h2(#installation). Set up Ably app and API Key
+
+We're going to need an API key to use when sending messages to our channel. In your "Ably dashboard":https://ably.com/accounts:
+
+# Hover over @Dashboard@ then select @Apps Dashboard@
+# Click @Create New App@
+# Name your new app
+# Copy the API Key somewhere safe
+
+h2(#webhook). Set up the webhook 
+
+Now we have an app we can set up a webhook to AWS Lambda to trigger when a new message published using the app. In your "Ably dashboard":https://ably.com/accounts:
```

Comment:
"set up a webhook to AWS Lambda" ... The Lambda IS the webhook. What you're setting up here is an integration rule that causes the Ably platform to make an HTTP request to the webhook every time a message is published.